File: Api/Dto/RecipeEditDto.cs
using System.ComponentModel.DataAnnotations;

namespace Api.Dto
{
    public record RecipeEditDto(
        int Id,
        [Required]
        [MaxLength(50)]
        string Title,
        [Required]
        [MaxLength(150)]
        string Description,
        [Required]
        [Range(1, 180)]
        int CookingMinutes,
        [Required]
        [Range(1, 100)]
        int NumberOfServings,
        string Image,
        [Required]
        int AuthorId,
        List<TagDto> TagDtos,
        List<IngredientDto> IngredientDtos,
        List<StepDto> StepDtos)
    {
    }
}

File: CullinaryBookSite/ClientApp/src/app/Dto/recipeEditDto.ts
import { IngredientDto } from "./ingredientDto";
import { StepDto } from "./stepDto";
import { TagDto } from "./tagDto";

export interface RecipeEditDto{
    id?: number;
    title: string;
    description: string;
    cookingMinutes: number;
    numberOfServings: number;
    image: Blob | null;
    authorId: number;
    tags: TagDto[];
    ingredients: IngredientDto[];
    steps: StepDto[];
}